frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -212,15 +212,22 @@ export const ClusterServiceVersionTableRow = withFallback<ClusterServiceVersionT
     const [icon] = _.get(obj, 'spec.icon', []);
     const deploymentName = _.get(obj, 'spec.install.spec.deployments[0].name');
     const namespace = getNamespace(obj);
+    const installPageRoute = `/operatorhub/install?pkg=${subscription.metadata.name}&catalog=${subscription.spec.source}&catalogNamespace=${subscription.spec.sourceNamespace}&targetNamespace=${namespace}&csvName=${obj.metadata.name}`;
     const route = resourceObjPath(obj, referenceFor(obj));
     const uid = getUID(obj);
     const internalObjects = getInternalObjects(obj);
+    const csvStatus = _.get(obj, 'status.phase', ClusterServiceVersionPhase.CSVPhaseUnknown);
     return (
       <TableRow id={uid} trKey={key} index={index} style={style}>
         {/* Name */}
         <TableData className={tableColumnClasses[0]}>
           <Link
-            to={route}
+            to={
+              csvStatus === ClusterServiceVersionPhase.CSVPhaseInstallReady ||
```

Comment:
Change this to an global array of accepted values and then do
`installPageRedirectStates.includes(csvStatus) ? installPageRoute : route `